lib/ansible/modules/remote_management/manageiq/manageiq_provider.py

```diff
@@ -0,0 +1,523 @@
+            default_role='default',
+            metrics_role=None,
+        ),
+    )
+
+
+def endpoint_list_spec():
+    return dict(
+        default=dict(required=True,
+                     type='dict', options=endpoint_argument_spec()),
+        metrics=dict(type='dict', options=endpoint_argument_spec()),
+    )
+
+
+def endpoint_argument_spec():
+    return dict(
+        role=dict(),
+        hostname=dict(required=True),
+        port=dict(required=True, type='int'),
+        verify_ssl=dict(default=True, type='bool'),
+        certificate_authority=dict(),
+        security_protocol=dict(
+            choices=[
+                'ssl-with-validation',
+                'ssl-with-validation-custom-ca',
+                'ssl-without-validation',
+            ],
+        ),
+        userid=dict(),
+        password=dict(no_log=True),
+        auth_key=dict(no_log=True),
+    )
+
+
+def delete_nulls(h):
+    """ Remove null entries from a hash
+
+    Returns:
+        a hash without nulls
+    """
+    if isinstance(h, list):
+        return map(delete_nulls, h)
+    if isinstance(h, dict):
+        return dict((k, delete_nulls(v)) for k, v in h.items() if v is not None)
+
+    return h
+
+
+class ManageIQProvider(object):
+    """
+        Object to execute provider management operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+
+    def class_name_to_type(self, class_name):
+        """ Convert class_name to type
+
+        Returns:
+            the type
+        """
+        out = [k for k, v in supported_providers().items() if v['class_name'] == class_name]
+        if len(out) == 1:
+            return out[0]
+
+        return None
+
+    def zone_id(self, name):
+        """ Search for zone id by zone name.
+
+        Returns:
+            the zone id, or send a module Fail signal if zone not found.
+        """
+        zone = self.manageiq.find_collection_resource_by('zones', name=name)
+        if not zone:  # zone doesn't exist
+            self.module.fail_json(
+                msg="zone %s does not exist in manageiq" % (name))
+
+        return zone['id']
+
+    def provider(self, name):
+        """ Search for provider object by name.
+
+        Returns:
+            the provider, or None if provider not found.
+        """
+        return self.manageiq.find_collection_resource_by('providers', name=name)
+
+    def build_endpoints(self, provider_type, raw_endpoints):
+        """ Build "connection_configurations" objects from
+        requested endpoints provided by user
+
+        Returns:
+            the user requested provider endpoints list
+        """
+        endpoints = []
+        provider_defaults = supported_providers().get(provider_type)
+
+        # build default endpoint
+        endpoint = raw_endpoints.get('default')
```

Comment:
👍 done, now using the variable "connection_configurations" instead of endpoints.